test/test_ssltransport.py

```diff
@@ -37,24 +51,43 @@ def sample_request(binary=True):
     return request if binary else request.decode("utf-8")
 
 
-def validate_request(provided_request, binary=True):
+def validate_request(
+    provided_request: bytearray, binary: Union["Literal[True]", "Literal[False]"] = True
+) -> None:
     assert provided_request is not None
     expected_request = sample_request(binary)
     assert provided_request == expected_request
 
 
-def sample_response(binary=True):
+@overload
+def sample_response(binary: "Literal[True]" = True) -> bytes:
+    ...
+
+
+@overload
+def sample_response(binary: "Literal[False]") -> str:
+    ...
+
+
+@overload
+def sample_response(binary: bool = False) -> Union[bytes, str]:
```

Comment:
Do we need this last `@overload` or is this covered by the function definition?

Reply:
I think yes. otherwise we will get:
```
test/test_ssltransport.py:86: error: No overload variant of "sample_response" matches argument type "bool"  [call-overload]
test/test_ssltransport.py:86: note: Possible overload variants:
test/test_ssltransport.py:86: note:     def sample_response(binary: Literal[True] = ...) -> bytes
test/test_ssltransport.py:86: note:     def sample_response(binary: Literal[False]) -> str

```

Reply:
I *think* that if you change here to `Literal[True, False]` it should work. That would mean that the function could only be called with a statically-known True or False, but since this is only used in tests, that seems like a good thing.

Reply:
@bluetech do you mean changing the last overload for the function? or changing `binary` variable type to `Literal[True, False]` in the function itself and removing the last overload?

Reply:
The latter. I'm not sure if it will work but worth a try.

Reply:
Thanks for the clarification. I've tried it but got the same error.
Let's keep it like this.